Lib/xmlrpc/client.py

```diff
@@ -1220,7 +1220,9 @@ def get_host_info(self, host):
         if isinstance(host, tuple):
             host, x509 = host
 
-        auth, host = urllib.parse._splituser(host)
+        p = urllib.parse.urlparse(host)
+        user, delim, host = p.path.rpartition('@')
```

Comment:
I don't see the benefit of avoid urllib to parse an URL. Parsing an URL is really hard, recently there were a lot of security issues reported (and fixed) in urllib. From my point of view, this change only makes xmlrpc less secure.

Reply:
@vstinner 
https://github.com/python/cpython/blob/8d0ef0b5edeae52960c7ed05ae8a12388324f87e/Lib/urllib/parse.py#L1039
I wrote this code since `urllib.parse.splituser()` recommends to use urlparse.
`_splituser` is a very simple code which is based on rpartition method.
So if this change makes xmlrpc less secure, How about revert it back to original code or get `auth` and `host` without using `_splituser`.

Reply:
There is no function in urllib.parse which can return username and password in two fields, rather than hardcoding '@'?

Reply:
@vstinner That's true. I've just rollback the code about _splituser.
IMHO, urllib.parse._splituser should be removed in the near future.
Because we deprecated `splituser` but end-user still able to use `splituser` by using `_splituser`
Any ideas to replace it?

Reply:
Anyway, I' ve now noticed that replacing `auth, host = urllib.parse._splituser(host)`to `urlparse` is definitely the wrong way.